codegen/src/test/resources/software/amazon/awssdk/codegen/poet/batchmanager/test-batchmanager-sync-class.java

```diff
@@ -0,0 +1,179 @@
+package software.amazon.awssdk.services.batchmanagertest.batchmanager.internal;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import software.amazon.awssdk.annotations.Generated;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.core.batchmanager.BatchManager;
+import software.amazon.awssdk.core.batchmanager.BatchOverrideConfiguration;
+import software.amazon.awssdk.services.batchmanagertest.BatchManagerTestClient;
+import software.amazon.awssdk.services.batchmanagertest.batchmanager.BatchManagerTestBatchManager;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestBatchResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.DeleteRequestResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestBatchResponse;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestRequest;
+import software.amazon.awssdk.services.batchmanagertest.model.SendRequestResponse;
+import software.amazon.awssdk.utils.ThreadFactoryBuilder;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkInternalApi
+public final class DefaultBatchManagerTestBatchManager implements BatchManagerTestBatchManager {
+    private boolean createdExecutor;
+
+    private final Executor executor;
+
+    private final BatchManagerTestClient client;
+
+    private final BatchManager<SendRequestRequest, SendRequestResponse, SendRequestBatchResponse> sendRequestBatchManager;
+
+    private final BatchManager<DeleteRequestRequest, DeleteRequestResponse, DeleteRequestBatchResponse> deleteRequestBatchManager;
+
+    private DefaultBatchManagerTestBatchManager(DefaultBuilder builder) {
+        this.client = builder.client;
+        ScheduledExecutorService scheduledExecutor = builder.scheduledExecutor;
+        ThreadFactory threadFactory = new ThreadFactoryBuilder().threadNamePrefix(
+            "software.amazon.awssdk.services.batchmanagertest.batchmanager.internal.DefaultBatchManagerTestBatchManager")
+                                                                .build();
```

Comment:
Can we move the threadFactory variable to createDefaultExectutor method? For the thread name prefix, how about something shorter, like `sdk-$serviceId(eg: sqs)-batchmanager`

Reply:
Moved threadFacotry to createDefaultExecutor method and changed thread name prefix as mentioned above.